File: Difficulty: Medium/Remove the balls/remove-the-balls.cpp
class Solution {
  public:
    int findLength(vector<int>& color, vector<int>& radius) {
        int n = color.size(); // Total number of balls

        stack<int> st;        // Stack to keep indices of non-removed balls

        for (int i = 0; i < n; i++) {
            // If top of stack has same color and radius → remove both
            if (!st.empty() && color[i] == color[st.top()] && radius[i] == radius[st.top()]) {
                st.pop();   // Remove the matched pair
            } else {
                st.push(i); // Otherwise, keep the current ball
            }
        }

        return st.size(); // Remaining balls after all removals
    }
};

File: Difficulty: Medium/Remove the balls/remove-the-balls.java
class Solution {
    public int findLength(int[] color, int[] radius) {
        int n = color.length;              // Total number of balls

        Stack<Integer> st = new Stack<>(); // Stack to keep indices of balls

        for (int i = 0; i < n; i++) {
            // Check if current ball matches the top of the stack
            if (!st.isEmpty() &&
                color[i] == color[st.peek()] &&
                radius[i] == radius[st.peek()]) {
                st.pop();   // Remove matching pair
            } else {
                st.push(i); // Keep current ball index
            }
        }

        return st.size(); // Remaining balls after all removals
    }
}